HardeningKitty.psm1

```diff
@@ -800,7 +800,13 @@
                             $Result = $Result -join ";"
                         }
                     } catch {
-                        $Result = $Finding.DefaultValue
+                        If ($Backup) {
+                            # If backup mode is enabled, with consider that this backup does not exists
+                            # and put "-NODATA-" as result to identify it as non-existing policy
+                            $Result = "-NODATA-"
+                        } Else {
+                            $Result = $Finding.DefaultValue
+                        }
                     }
                 } Else {
                     $Result = $Finding.DefaultValue
```

Comment:
If the registry path does not exist (which will be the case for all *...Policy...* paths) then "-NODATA-" applies here as well

Reply:
Yes.. And No. Yes, and if policies does not exist on backup,  they should be removed when this backup is applied with HailMary mode. No because not all policies have a non-existent path, and some of them (from CIS for example) are already exist in OS registry. I don't understand what do you mean exactly ? Should I remove this modification ?

Reply:
Au contraire, we need an additional check if `Test-Path -Path $Finding.RegistryPath` fails (line 794) or else the default value is stored for an non existing path

Reply:
Oh yes ! Good point ! I will fix it ;)